core/cas-server-core-configuration-api/src/main/java/org/apereo/cas/configuration/DefaultCasConfigurationPropertiesSourceLocator.java

```diff
@@ -39,37 +33,67 @@
 @Slf4j
 @RequiredArgsConstructor
 public class DefaultCasConfigurationPropertiesSourceLocator implements CasConfigurationPropertiesSourceLocator {
+    // Extensions will be processed in order, last one wins
+    private static final String[] EXTENSIONS = {"properties", "yml", "yaml"};
+
     private final CipherExecutor<String, String> configurationCipherExecutor;
     private final CasConfigurationPropertiesEnvironmentManager casConfigurationPropertiesEnvironmentManager;
 
-    private static Collection<File> scanForConfigurationFilesByPattern(final File config, final String regex) {
-        return FileUtils.listFiles(config, new RegexFileFilter(regex, IOCase.INSENSITIVE), TrueFileFilter.INSTANCE)
-            .stream()
-            .sorted(Comparator.comparing(File::getName))
-            .collect(Collectors.toList());
+
+    // will be processed in order, last one wins
+    private List<String> getApplicationNames() {
+        val appNames = new ArrayList<String>();
+        appNames.add("application");
+        appNames.add(casConfigurationPropertiesEnvironmentManager.getApplicationName());
+        return appNames;
     }
 
-    private static String buildPatternForConfigurationFileDiscovery(final File config, final List<String> profiles) {
-        val propertyNames = String.join("|", profiles);
-        val profiledProperties = profiles.stream()
-            .map(p -> String.format("application-%s", p))
-            .collect(Collectors.joining("|"));
+    // will be processed in order, last one wins
+    private List<File> getAllPossibleExternalConfigDirFilenames(final File configdir, final List<String> profiles) {
+        val fileNames = new ArrayList<File>();
+        for (val appName: getApplicationNames()) {
```

Comment:
Opportunity to use the Java stream API?

Reply:
If by opportunity you mean opportunity for me to learn the streams api better, then yes, let's call it an opportunity... but after converting it to streams or in the process of doing so I noticed in the debugger that I was looking for CAS.properties which was working on windows but not on linux (since the file is named cas.properties in the unit tests) so that is why test was working for me locally and not on in travis.

Reply:
I usually cheat myself, and write the loops and whatnot in the old traditional way and then use IDEA's inspections to convert things back to the fancy syntax. That's been a very valuable tool for me to learn the API and over time I'd gotten better at it.